### PR TITLE
Add link to public questions page to sharing admin page

### LINKED ISSUE #9397: Question Sharing: Path out of Beta
I had a meeting with @nwalters512 and @mwest1066, and we discussed 3 needed features to get 

1. A link on the Instructor Sharing Settings page to the course's public sharing page
2. In the sidebar of a question preview, the QID that the question is shared as (which is @course-sharing-name/qid)
3. A way for people to copy the code of shared questions (if the authors desire it).

The first 2 are easy, number 3 has quite a bit of stuff to unpack, so I'll lay out a bit more of the details that we discussed: 

on the Questions Settings page:
- When someone goes to 'publicly share' a question , they should have a checkbox where they can state if they want it to be source available under some CC license (probably only one choice, allowing more choices will be too much trouble. we discussed using CC-BY)
- After publicly sharing, they can hit a button that says 'edit public sharing' or similar so they can either un-share their source by unchecking the box, or check the box to share the source if they hadn't initially (we can't allow people to un-share questions at risk of breaking existing assessments, but we can allow people to un-share source without any negative repercussions)
- add another badge on the page saying if the source is shared

In the JSON: 
- add an 'authors' field to the info.json for questions. This is just a list of emails. We will destructure this into the database so that in the future we can do joins on the email in order to give people usage statistics for their questions based on their email. In the future we may allow adding syntax for people to put their name with their email (e.g. "First Last <name@provider.com>")

In the DB: 
- a new field on questions, 'publicly_shared_source'
- the new 'question_authors' table to mirror the JSON change described above

--------------------------------------

Edit: Adding a TODO list here of all the items mentioned in this thread so we can track completion: 
- [x] A link on the Instructor Sharing Settings page to the course's public sharing page (#9478)
- [x]  In the sidebar of a question preview, the QID that the question is shared as (which is @course-sharing-name/qid) (#9763)
- [ ] authors field in JSON and database (#9790)
- [x] copy button on public shared question previews (#9800)
- [x] sharing course should get a sync error if they try to delete or change the QID of a question that they have shared (#9797)
- [x] limit expensive features (at least workspaces) from publicly-viewable questions (#9902)
- [x] Make sure custom course elements that have client assets (JS, CSS, etc.) work properly
  - [x] publicly shared question previews (#9807)
  - [x] shared questions being used within a course (#9786)
- [x] Fix 'undefined' in taskbar on public question previews (#9094)
- [ ] Move declaration of sharing info for questions into JSON (sharedPublicly, sharingSets, sharedPubliclyWithSource) (#10029)
- [ ] More clear warnings about permissions being granted (for client element files and potentially clientFilesCourse) (#10730)
  - [ ] In the question sharing documentation page
  - [ ] In the  "Add Course to Sharing Set" popover (make it a dialog) on the sharing page
  - [ ] In the "Share publicly" dialog on the question settings page

### PR DESCRIPTION
**Pull request by SethPoulsen**
Resolves part of #9397

### CODE REVIEW

**Comment by SethPoulsen** [comment]
image

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts [hunk 1]
```diff
@@ -131,6 +136,29 @@ export const InstructorSharing = ({
...
+              window.setTimeout(function () {
+                $(e.trigger).popover('hide');
+              }, 1000);
+            });
+            $('.js-public-sharing-link-qrcode-button').popover({
+              content: $('#js-public-sharing-link-qrcode'),
+              html: true,
+              trigger: 'click',
+              container: 'body',
+            });
```
Comment (SethPoulsen) [inline apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts:159]: I copied this from the Assessment settings page, in the interest of making the 'copy this link/qr code' UI look similar. But I'm not sure if that was a good idea. Also, we now have two different copy buttons on this page which look entirely different from one another.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts [hunk 2]
```diff
@@ -190,6 +218,37 @@ export const InstructorSharing = ({
...
+                        <button
+                          type="button"
+                          title="Public Questions Link QR Code"
+                          aria-label="Public Questions QR Code"
+                          class="btn btn-sm btn-outline-secondary js-public-sharing-link-qrcode-button"
+                        >
+                          <i class="fas fa-qrcode"></i>
+                        </button>
+                        <div class="d-none">
+                          <div id="js-public-sharing-link-qrcode">${publicSharingLinkQRCode}</div>
```
Comment (SethPoulsen) [inline apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts:246]: For some reason, this QR code SVG is not getting un-escaped properly, and it ends up just being text in the browser instead of rendering as SVG. Does this have something to do with something I don't understand about the prairielearn/html package?

**Comment by mwest1066** [comment]
> Partially resolves #9397

Note that GH only looks at the "resolves #XXX" part of this and will thus close that issue automatically. I'd suggest rewording as "resolves part of #XXX" or similar.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts:159]: I think matching the other copy button on this page would be better. The QR code on other pages is meant for lectures/labs where this will be projected up for students to scan, but I don't think there's currently any use case for that for the public sharing page.

**Review by mwest1066 — APPROVED** [review]

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts]: QR code removed in 0385fb4; all copy buttons on the page now match.

File: apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts (hunk 2 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorCourseAdminSharing/instructorCourseAdminSharing.html.ts]: QR code removed in 0385fb4.
